src/xrGame/xrGame.cpp

```diff
@@ -78,11 +78,14 @@ static void load(int argc, char** argv, char** envp)
 // XXX nitrocaster PROFILER: temporarily disabled due to linkage issues
 // g_profiler			= new CProfiler();
 #endif
+    gStringTable = new CStringTable();
```

Comment:
__attribute__((constructor)) and __attribute__((destructor)) will not be accepted by msvc, and will still be under #ifdef. To be honest, I just don’t know how to do it beautifully.

Reply:
I am not sure about the behaviour of the compiler on Linux, but what about IFDEF'ing only the lines with __attribute (and leaving the function body not IFDEF'ed)? It would allow us to call the specified functions from DllMain on Windows instead of duplicating the code. At least such solution relieves us from maintaining sync for some parts of the code in case of future changes.

Reply:
You can replace `__attribute__((constructor))` with [[gnu::constructor]] to make MSVC accept those functions. Maybe it prints an "unknown attribute" warning, but it won't be an error.